tests/Models/Post.php

```diff
@@ -34,9 +34,19 @@ public function getCreator(): ActiveQuery
      *
      * @return ActiveQuery<User>
      */
-    public function getContributors(): AvtiveQuery
+    public function getContributors(): ActiveQuery
     {
         return $this->hasMany(User::class, ['user_id' => 'user_id']);
     }
 
+    /**
+     * Gets the user that created this post
+     *
+     * @return CategoryQuery<Category>
+     */
+    public function getCategory(): CategoryQuery
```

Comment:
Why does this return a CategoryQuery, when `getContributors()` returns an ActiveQuery? Should `getContributors()` be more precise, or should `getCategory()` be more generic?

Reply:
This is testing that you can use the functionality with custom queries. The `Category` model has an overridden `find` method that returns an [`CategoryQuery`](https://github.com/Practically/PsalmPluginYii2/blob/feat/magic-props/tests/Models/Category.php#L26). All the other models don't override the `find` function, so they return an `ActiveQuery`